dev/com.ibm.ws.kernel.service/src/com/ibm/ws/common/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,382 @@
+    /**
+     * List of supported Message Digest Algorithms.
+     */
+    private static final List<String> supportedMessageDigestAlgorithms = Arrays.asList(
+                                                                                       MESSAGE_DIGEST_ALGORITHM_SHA256,
+                                                                                       MESSAGE_DIGEST_ALGORITHM_SHA384,
+                                                                                       MESSAGE_DIGEST_ALGORITHM_SHA512);
+
+    public static String getMessageDigestAlgorithm() {
+        return MESSAGE_DIGEST_ALGORITHM_SHA256;
+    }
+
+    public static MessageDigest getMessageDigest() throws NoSuchAlgorithmException {
+        return getMessageDigest(getMessageDigestAlgorithm());
+    }
+
+    public static MessageDigest getMessageDigest(String algorithm) throws NoSuchAlgorithmException {
+        if (!supportedMessageDigestAlgorithms.contains(algorithm))
+            throw new NoSuchAlgorithmException(String.format("Algorithm %s is not supported", algorithm));
+        return MessageDigest.getInstance(algorithm);
+    }
+
+    public static MessageDigest getMessageDigestForLTPA() {
+        MessageDigest md1 = null;
+        try {
+            if (fipsEnabled) {
+                if (isSemeruFips()) {
+                    md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA256,
+                                                    OPENJCE_PLUS_FIPS_NAME);
+                } else {
+                    md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA256,
+                                                    IBMJCE_PLUS_FIPS_NAME);
+                }
+            } else {
+                md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA);
+            }
+
+        } catch (NoSuchAlgorithmException e) {
+            // instrumented ffdc
+        } catch (NoSuchProviderException e) {
+            // instrumented ffdc;
+        }
+
+        return md1;
+    }
+
+    public static String getProperty(final String prop, final String defaultValue) {
+        return AccessController.doPrivileged(new PrivilegedAction<String>() {
+            @Override
+            public String run() {
+                return System.getProperty(prop, defaultValue).toLowerCase();
+            }
+        });
+    }
+
+    static String getFipsLevel() {
+        return getProperty("com.ibm.fips.mode", "disabled");
+    }
+
+    public static boolean isSemeruFips() {
+        return "true".equals(getProperty("semeru.fips", "false"));
+    }
+
+    public static boolean isFips140_3Enabled() {
+
+        return isRunningBetaMode() &&
+               "140-3".equals(FIPSLevel) || "true".equalsIgnoreCase(getProperty("global.fips_140-3", "false")) || isSemeruFips();
+    }
+
+    public static boolean isFips140_2Enabled() {
+        return isRunningBetaMode() && "140-2".equals(FIPSLevel);
```

Comment:
Don't we already have 140-2 support?  Won't this break that?

Reply:
140-2 is baked in with the code ... we don't have a flag for 140-2 yet. ...